test/framework/src/test/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertionsTests.java

```diff
@@ -52,4 +59,172 @@ public void writeTo(StreamOutput out) throws IOException {
             throw new IllegalArgumentException("Not supported.");
         }
     }
+
+    public void testAssertXContentEquivalent() throws IOException {
```

Comment:
After using RandomObjects.randomStoredFieldValues this should also test bytes[] now.